### PR TITLE
Add submission count badge to quick-eval widget

### PR DESCRIPTION
**Pull request by Scy-D2L**
Added a submission-count badge to the icon in the quick-eval activities list
image

- Added the `d2l-quick-eval-widget-submission-icon` component temporarily. After discussing with Gaudi there should be a similar generic badge component coming to core soon, and we should replace it when it does.
- Fixed a defect in `activity-list-item-basic` that prevented the date and course names from being read by screenreaders

### CODE REVIEW

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-basic.js [hunk 1]
```diff
@@ -160,18 +161,24 @@ class ActivityListItemBasic extends ListItemLinkMixin(SkeletonMixin(EntityMixinL
...
-				<d2l-icon
-					class=${classMap(iconClasses)}
-					?skeleton=${this.skeleton}
-					icon=${this._icon}>
-				</d2l-icon>`,
+			illustration: this.submissionCount ? html`
+					<d2l-quick-eval-widget-submission-icon style="overflow: visible;"
+						class="class=${classMap(iconClasses)}"
+						icon=${this._icon}
+						submission-count=${this.submissionCount > 99 ? '99+' : this.submissionCount}>
```
Comment (Scy-D2L) [inline components/d2l-work-to-do/d2l-work-to-do-activity-list-item-basic.js:168]: This is one point of minor concern - are there any RTL / alternative language pieces that make hardcoding `99+` here a problem?

File: components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+					right: auto;
+				}
+				d2l-icon {
+					height: 1.3rem;
+					width: 1.3rem;
+				}
+		`];
+	}
+
+	constructor() {
```
Comment (vix633) [inline components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js:86]: Do we need the empty ctor?

File: components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (!changedProperties.has('submissionCount')) return;
+		this._submissionCountHidden = !(this.submissionCount && this.submissionCount.length > 0);
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('a');
```
Comment (dbatiste) [inline components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js:108]: Looks like this component doesn't render an `a`.

Comment (Scy-D2L) [reply components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js]: Thanks, looks like I missed a few things making the trimmed down icon from `card-footer-link`.

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-basic.js (hunk 1 above)
Comment (dbatiste) [reply components/d2l-work-to-do/d2l-work-to-do-activity-list-item-basic.js:168]: Most times it stays exactly the same, but there appears to be a couple of translations for this.

ex. https://search.d2l.dev/xref/GitHub/Brightspace/folio-app/src/lang/ar.json?r=3d2d8f52#335

File: components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: Preset icon key (e.g. "tier1:gear")
+			 */
+			icon: { type: String, reflect: true },
+			/**
+			 * Submission count to display as a superscript on the icon
+			 */
+			submissionCount: { type: String, attribute: 'submission-count', reflect: true }
```
Comment (dbatiste) [inline components/d2l-quick-eval-widget/d2l-quick-eval-widget-submission-icon.js:22]: FYI - when the time comes to provide this in core, we will likely retain the original property name `secondaryText` and `secondaryTextType (count|notification)`, unless there are any use cases that are discovered that would affect that.

**Review by dbatiste — APPROVED** [review]
Aside from potential need for translation, looks good.

**Pull request merged**